lib/engine/game/g_18_fr/train.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G18FR
+      class Train < Engine::Train
+        attr_reader :base_price, :final_price
+
+        def initialize(name:, distance:, price:, index: 0, **opts)
+          super
+
+          @base_price = price
+          @final_price = opts[:final_price] || price
+        end
+
+        def new_price(price)
```

Comment:
You could do this as a setter method:
```suggestion
        def price=(price)
```
That would mean you would be able to say `train.price = xxx` instead of `train.new_price(xxx)` which feels a bit more natural to me.